pkg/docker/docker.go

```diff
@@ -163,6 +171,60 @@ func New(config *api.DockerConfig, auth docker.AuthConfiguration) (Docker, error
 	}, nil
 }
 
+// StartContainer starts an existing container.
+func (d *stiDocker) StartContainer(id string, hostConfig *docker.HostConfig) error {
+	return d.client.StartContainer(id, hostConfig)
+}
+
+// KillContainerOptions kills the container using SIGKILL.
+func (d *stiDocker) KillContainer(containerID string) error {
+	opts := docker.KillContainerOptions{ID: containerID}
+	return d.client.KillContainer(opts)
+}
+
+// dockerExec executes the command in specified container
+func (d *stiDocker) dockerExec(containerName string, command []string) error {
+	config := docker.CreateExecOptions{
+		Container:    containerName,
+		AttachStdin:  true,
+		AttachStdout: true,
+		AttachStderr: false,
+		Tty:          false,
+		Cmd:          command,
+		User:         "root",
+	}
+	execCmd, err := d.client.CreateExec(config)
+	if err != nil {
+		return err
+	}
+	var stdout, stderr bytes.Buffer
+	success := make(chan struct{})
+	opts := docker.StartExecOptions{
+		OutputStream: &stdout,
+		ErrorStream:  &stderr,
+		RawTerminal:  true,
+		Success:      success,
+	}
+	go func() {
+		err = d.client.StartExec(execCmd.ID, opts)
+	}()
+	<-success
+	glog.V(5).Infof("Executed %#v command in %q", command, containerName)
+	return err
+}
+
+// ChangeContainerFileOwner changes an owner of a single file inside the
+// container.
+func (d *stiDocker) ChangeContainerFilesOwner(containerName, username string, files []string) error {
+	return d.dockerExec(containerName, append([]string{"chown", "-R", "-c", username}, files...))
```

Comment:
same concern @csrwng raised about rm... this chown command is being run as root.  if the user overrode what "chown" is in their image, we just gave them the ability to run arbitrary root operations in the container.